src/LightSpeed/controlpanel.js

```diff
@@ -120,6 +125,7 @@ var LSCombinedTopic = GObject.registerClass({
             if (!(e instanceof SyntaxError || e instanceof ReferenceError))
                 throw e;
             this._variablesCodeview.setCompileResultsFromException(e);
+            this.set_property('needs-attention', true);
```

Comment:
TIL that `g_object_set_property()` sets the property _and_ emits the notify signal automatically! I will really have to rethink how I usually write GObject properties in GJS now.

I was going to suggest defining setter and getter functions like in the TopicButton class. You could still consider doing that, to make it so that the notify signal isn't emitted if the property already has the correct value, but on reflection I think it's probably fine as is.

Reply:
Yeah, I figured that we're doing something with the signal that's pretty inexpensive and we have the check in the set() for the topicButton. In general I agree that having the explicit methods is better.